### PR TITLE
Change note for non marked with main actor global variables

### LINKED ISSUE #74751: Presentation different for identical concurrency note
### Motivation

I found two situations that are recommending the same thing, but in different ways.

input 1:

```swift
var global = 42
```

diagnostics 1:

```
test.swift:1:5: error: var 'global' is not concurrency-safe because it is non-isolated global shared mutable state
1 | var global = 42
  |     |- error: var 'global' is not concurrency-safe because it is non-isolated global shared mutable state
  |     |- note: convert 'global' to a 'let' constant to make 'Sendable' shared state immutable
  |     |- note: annotate 'global' with '@MainActor' if property should only be accessed from the main actor
  |     `- note: disable concurrency-safety checks if accesses are protected by an external synchronization mechanism
2 | 
```

Highlight:

`note: annotate 'global' with '@MainActor' if property should only be accessed from the main actor`

input 2:

```swift
var global = 42

func mutate() {
    global += 1
}
```

diagnostics 2:

```
test.swift:4:5: error: main actor-isolated var 'global' can not be mutated from a non-isolated context
1 | var global = 42
  |     `- note: mutation of this var is only permitted within the actor
2 | 
3 | func mutate() {
  |      `- note: add '@MainActor' to make global function 'mutate()' part of global actor 'MainActor'
4 |     global += 1
  |     `- error: main actor-isolated var 'global' can not be mutated from a non-isolated context
5 | }
6 | 
```

highlight: `note: add '@MainActor' to make global function 'mutate()' part of global actor 'MainActor'`

### Proposed solution

I think the second note seems like its more general-purpose. But, I think including "annotate" does help.

`note: annotate 'global' with '@MainActor' if property should only be accessed from the main actor`

could become:

`note: add '@MainActor' annotation to make property 'global' part of global actor 'MainActor'`

### Alternatives considered

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by rofle100lvl**
Resolves #74751
- Changed note
- Changed tests

### CODE REVIEW

**Comment by rofle100lvl** [comment]
@swift-ci please test

File: include/swift/AST/DiagnosticsSema.def [hunk 1]
```diff
@@ -5646,7 +5646,7 @@ NOTE(shared_state_make_immutable,none,
      "convert %0 to a 'let' constant to make 'Sendable' shared state immutable",
      (const ValueDecl *))
 NOTE(shared_state_main_actor_node,none,
-     "annotate %0 with '@MainActor' if property should only be accessed from the main actor", (const ValueDecl *))
+     "add '@MainActor' annotation to make property %0 part of global actor 'MainActor'", (const ValueDecl *))
```
Comment (hborla) [inline include/swift/AST/DiagnosticsSema.def:5649]: I gathered from the GitHub issue that there already is a note that says almost this exact text. It's in this same DiagnosticsSema.def file:

```
NOTE(add_globalactor_to_function,none,
     "add '@%0' to make %kind1 part of global actor %2",
     (StringRef, const ValueDecl *, Type))
```

This note already takes arguments, so we can pass in `MainActor` as both the string and the type. I think we can just remove `shared_state_main_actor_node`, and use `add_globalactor_to_function`.

**Review by hborla — COMMENTED** [review]
Thank you for your contribution! I left a suggestion inline

File: include/swift/AST/DiagnosticsSema.def (hunk 1 above)
Comment (rofle100lvl) [reply include/swift/AST/DiagnosticsSema.def:5649]: Hi, @hborla. It does make sense for me, however I do not understand how to get MainActorType to pass it as last parameter in add_globalactor_to_function. I tried getASTContext().getMainActorType() to variables in function, but it always retrun null for me. Can you give a suggestion please?

Comment (rofle100lvl) [reply include/swift/AST/DiagnosticsSema.def:5649]: @hborla  I tried getASTContext().getMainActorType() with different VarDecl variables and it always return null, but for functions it doesn't return null. Feels weird.
Also tried that approaches:
1) Isolation.getGlobalActor -> But the variant doesn't have any actor so I get assert
2) SwiftLang.getMainActor, but it is located in clang and I think that it shouldn't be used in AST module.

Comment (hborla) [reply include/swift/AST/DiagnosticsSema.def:5649]: Hmm, could you show me what you tried? This works for me locally in `ActorIsolationRequest::evaluate` at the point where it was previously diagnosing `diag::shared_state_main_actor_node` on line 5459:

```swift
          auto mainActor = ctx.getMainActorType();
          if (!mainActor)
            return isolation;

          diagVar->diagnose(diag::add_globalactor_to_function,
              mainActor->getWithoutParens().getString(), diagVar, mainActor)
              .fixItInsert(diagVar->getAttributeInsertionLoc(false),
                           diag::insert_globalactor_attr, mainActor);
```

I also needed to add a capture of `&ctx` to the lambda capture list.

Comment (rofle100lvl) [reply include/swift/AST/DiagnosticsSema.def]: This code perfectly work. You can't imagine how many times I tried it on my local machine, but somewhy when I launch it through Xcode it's always fails(even on my second laptop). But on third laptop it works, kind of magic. Now I tried to launch tests and it works fine on all the machines
> ```swift
>           auto mainActor = ctx.getMainActorType();
>           if (!mainActor)
>             return isolation;
> 
>           diagVar->diagnose(diag::add_globalactor_to_function,
>               mainActor->getWithoutParens().getString(), diagVar, mainActor)
>               .fixItInsert(diagVar->getAttributeInsertionLoc(false),
>                            diag::insert_globalactor_attr, mainActor);
> ```

Thank you for your help @hborla

**Review by hborla — APPROVED** [review]
Thank you!

**Comment by hborla** [comment]
@swift-ci please smoke test

**Pull request merged**